Do-It/Controller/DueDatePickerTableViewController.swift

```diff
@@ -0,0 +1,155 @@
+//
+//  DueDatePickerTableViewController.swift
+//  Do-It
+//
+//  Created by Aaron on 2/21/19.
+//  Copyright © 2019 The Swifter Picker-Uppers. All rights reserved.
+//
+
+import UIKit
+
+protocol DueDatePickerTableViewControllerDelegate: AnyObject {
+    func dueDatePickerTableViewController(_ viewController: DueDatePickerTableViewController,
+                                          didSelectTimeRange range: DateInterval?)
+}
+
+class DueDatePickerTableViewController: UITableViewController {
+
+    var timeRangeSelected: DateInterval?
+
+    struct Option {
+
+        let index: Int
+        let name: String
+        let dateRange: DateInterval
```

Comment:
We want to be able to encode these date intervals as being relative to the current date, so that if this filter date range is persisted as the user's settings, it isn't defined as a specific period of time.

To do this, we can store DateComponents that contain the relative offset. For example, to encode a 'within one week' range, all we need is a DateComponents object with its `day` field set to 7.